ASAA-28 Adding PR List to Home Screen

LuisRocketRangel opened the pull request:
Requirements:
-Add a Pull Request List to Home Screen
-Add a new Pull Request Card to Match Zeplin

Solution:
-Added a new Pull Request Card Composable
-Added a new LazyColumn to have Pull Requests i
-Made the Home Screen Scrollable

![Screenshot_20220927-142321](https://user-images.githubusercontent.com/83731304/192621894-dac32e83-5b80-4101-9ad4-deebb821740e.png)
![Screenshot_20220927-142330](https://user-images.githubusercontent.com/83731304/192621903-e8d0e2f3-cbd3-4961-b826-8a51bdb068c7.png)

LuisRocketRangel committed:
ASAA-28 Adding PR List to Home Screen

Requirements:
-Add a Pull Request List to Home Screen
-Add a new Pull Request Card to Match Zeplin

Solution:
-Added a new Pull Request Card Composable
-Added a new LazyColumn to have Pull Requests in
-Made the Home Screen Scrollable

IvanTelisman commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/home/HomeViewModel.kt:
```diff
@@ -20,8 +22,28 @@ class HomeViewModel : BaseViewModel() {
 
     // UI
+
+    val userPullRequestState: Flow<List<PullRequestItemState>> =
```
@LuisRocketRangel NIT: Not a huge deal but you could probably use a UI model object here similar to the UserRepositoryUiModel being used below, with the idea being that you could just only add the PR data that you need to be displayed in the card into that UI model object.

IvanTelisman commented on compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt:
```diff
@@ -88,6 +98,62 @@ fun CardLayout(userRepositoryUiModel: UserRepositoryUiModel, selectItem: (userRe
 }
 
+@Composable
+fun HomePullRequestItemCard(state: PullRequestItemState) {
```
@LuisRocketRangel NIT: super minor and it looks like it might be a little tricky to reuse CardLayout above this for this PR card layout but maybe just to make things a little easier to read we can rename CardLayout to RepositoryCardLayout and then rename HomePullRequestItemCard to PullRequestCardLayout.

LuisRocketRangel replied:
Will fix!

LuisRocketRangel committed:
ASAA-28 PR Feedback

Requirements:
-Rename some composables
-Use a UIModel Item for Pull Requests

Solution:
-Did above

IvanTelisman reviewed (approved):
Looks good @LuisRocketRangel ! Nice work and thanks for the screen shots too!

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/pullrequests/PullRequestViewModel.kt:
```diff
@@ -27,9 +27,13 @@ class PullRequestViewModel : BaseViewModel() {
                 linesAdded = "0 Lines Added".asMutableState(),
-                linesRemoved = "0 Lines Removed".asMutableState()
+                linesRemoved = "0 Lines Removed".asMutableState(),
+                reviewers = "No Reviewers".asMutableState(),
```
@LuisRocketRangel thanks for adding the trailing comma here :)

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt:
```diff
@@ -17,17 +20,23 @@ import androidx.compose.ui.Alignment
 
 @Composable
-fun CardLayout(userRepositoryUiModel: UserRepositoryUiModel, selectItem: (userRepositoryUiModel: UserRepositoryUiModel) -> Unit) {
+fun RepositoryCardLayout(userRepositoryUiModel: UserRepositoryUiModel, selectItem: (userRepositoryUiModel: UserRepositoryUiModel) -> Unit) {
```
@LuisRocketRangel Appreciate the name refactor here!

jaredstockton committed:
ASAA-28 Review commit: Unicode related tweaks

* feat: Add brCustomAndroidLintRules support to compose module to prevent unicode characters directly being added to source files
* bug: Replace rendered unicode bullet and right arrow characters with their numerical representations to prevent lint failures

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt:
```diff
@@ -88,6 +97,63 @@ fun CardLayout(userRepositoryUiModel: UserRepositoryUiModel, selectItem: (userRe
+                            .background(color = brown_grey, shape = RoundedCornerShape(Dimens.grid_0_5))
+                            .padding(Dimens.grid_0_75)
+                    )
+                    Text(text = "\u2192", style = typography.body1)
```
@LuisRocketRangel No fault of your own but I did want to mention some recent changes I've made :)

Note that I've replaced the rendered unicode right arrow character with the numeric representation. This **should** have been caught when running the Android Studio `lintInternalDebug` task locally **but wasn't because the BR custom android lint rules weren't added to that module** (until now).

Overall, I just pushed a review commit that adds the custom lint rules to the compose module and replaced the rendered unicode characters with the numeric representations.

More info on why those rendered unicode characters aren't allowed in source code at https://github.com/BottleRocketStudios/Android-CustomLintRules (integrated into the arch demo in https://github.com/BottleRocketStudios/Android-ArchitectureDemo/pull/41)

jaredstockton reviewed (commented):
Early review comments (apologies for getting to these late). Still going through this :)

jaredstockton reviewed (approved):
@LuisRocketRangel Everything else looks good! Thanks for the contribution :)
